core/domain/user_services.py

```diff
@@ -511,85 +511,92 @@ def get_all_user_groups() -> List[user_domain.UserGroup]:
         user_models.UserGroupModel.get_all())
     user_groups_list = []
     for user_group_model in user_group_models:
-        user_usernames: List[str] = []
-        for user_setting in get_users_settings(user_group_model.user_ids):
-            assert user_setting is not None
-            assert user_setting.username is not None
-            user_usernames.append(user_setting.username)
+        member_usernames: List[str] = []
+        user_ids = user_group_model.user_ids
+        user_settings_list = get_users_settings(user_ids, strict=True)
+        for user_id, user_settings in zip(user_ids, user_settings_list):
```

Comment:
Thinking about this a bit more -- let's say we create a user group with user X in it, then user X deletes their account (in the self-service way through the preferences page). How is this handled when we next reload the user groups page? Is there a backend test that covers this scenario?

Reply:
Right, make sense. This part is being handled in the wipeout_services.py, when we delete the user it will be deleted from all the UserGroupModel objects.
The test have been added in the wipeout_services_test.py file.